lib/ImageData.js

```diff
@@ -161,7 +161,12 @@ class ImageData {
         const suffix = fileSuffix || "";
         const fileName = path.parse(this.baseName).name;
         const extension = "." + this.type.ext;
-        if ( directory != null ) {
+
+        if ( typeof directory === "object" && directory.regex && directory.regex.find ) {
+            const regex = new RegExp(directory.regex.find);
+            const output = this.dirName.replace(regex, directory.regex.replace || "");
+            return path.join(output, prefix + fileName + suffix + extension);
```

Comment:
I think `RegExp` is bad idea a bit.

Because:
- Now configuration is `json` format. so we have to write regexp as string.
- Configuration seems to become complicated.